### PR TITLE
Fix up array list<> docblocks.

### PR DESCRIPTION
**Pull request by dereuromark**
What do you think about fixing up the non assoc array docs

A few things I notived:

- App::path() apparently still also returns assoc, I wonder if we want to remove this in 6.0?

### CODE REVIEW

**Comment by dereuromark** [comment]
I dont quite get what this weird thing is

    public function getHeaders(array $include = []): array
    {
        $this->createBoundary();

        if ($include === array_values($include)) {
            $include = array_fill_keys($include, true);
        }
        
I had to revert the list<> here for now.

File: src/Core/functions.php [hunk 1]
```diff
@@ -110,7 +110,7 @@ function pluginSplit(string $name, bool $dotAppend = false, ?string $plugin = nu
      * Commonly used like `list($namespace, $className) = namespaceSplit($class);`.
      *
      * @param string $class The full class name, ie `Cake\Core\App`.
-     * @return array<string> Array with 2 indexes. 0 => namespace, 1 => classname.
+     * @return array<int, string> Array with 2 indexes. 0 => namespace, 1 => classname.
```
Comment (ADmad) [inline src/Core/functions.php]:
```suggestion
     * @return array{0: string, 1: string} Array with 2 indexes. 0 => namespace, 1 => classname.
```

File: src/Core/functions_global.php [hunk 2]
```diff
@@ -75,7 +75,7 @@ function pluginSplit(string $name, bool $dotAppend = false, ?string $plugin = nu
      * Commonly used like `list($namespace, $className) = namespaceSplit($class);`.
      *
      * @param string $class The full class name, ie `Cake\Core\App`.
-     * @return array<string> Array with 2 indexes. 0 => namespace, 1 => classname.
+     * @return array<int, string> Array with 2 indexes. 0 => namespace, 1 => classname.
```
Comment (ADmad) [inline src/Core/functions_global.php]:
```suggestion
     * @return array{0: string, 1: string} Array with 2 indexes. 0 => namespace, 1 => classname.
```

File: src/Database/Driver/TupleComparisonTranslatorTrait.php [hunk 3]
```diff
@@ -71,7 +71,7 @@ protected function _transformTupleComparison(TupleComparison $expression, Query
         $true = new QueryExpression('1');
 
         if ($value instanceof SelectQuery) {
-            /** @var array<string> $selected */
+            /** @var array<int, string> $selected */
```
Comment (ADmad) [inline src/Database/Driver/TupleComparisonTranslatorTrait.php]:
```suggestion
            /** @var list<string> $selected */
```

**Pull request merged**